dotnet/src/Experimental/Process.LocalRuntime/LocalStep.cs

```diff
@@ -39,47 +40,37 @@ internal class LocalStep : KernelProcessMessageChannel
     /// <summary>
     /// Represents a step in a process that is running in-process.
     /// </summary>
-    /// <param name="name">Required. The name of the step.</param>
-    /// <param name="id">Required. The unique Id of the step.</param>
+    /// <param name="stepInfo">An instance of <see cref="KernelProcessStepInfo"/></param>
     /// <param name="kernel">Required. An instance of <see cref="Kernel"/>.</param>
     /// <param name="parentProcessId">Optional. The Id of the parent process if one exists.</param>
     /// <param name="loggerFactory">An instance of <see cref="LoggerFactory"/> used to create loggers.</param>
-    public LocalStep(string name, string id, Kernel kernel, string? parentProcessId = null, ILoggerFactory? loggerFactory = null)
+    public LocalStep(KernelProcessStepInfo stepInfo, Kernel kernel, string? parentProcessId = null, ILoggerFactory? loggerFactory = null)
     {
-        Verify.NotNullOrWhiteSpace(name);
-        Verify.NotNullOrWhiteSpace(id);
+        Verify.NotNull(stepInfo);
         Verify.NotNull(kernel);
+        Verify.NotNullOrWhiteSpace(stepInfo.State.Name);
 
-        this.Name = name;
-        this.Id = id;
+        this.Id = stepInfo.State.Id ?? Guid.NewGuid().ToString("n");
+        this.Name = stepInfo.State.Name;
```

Comment:
A reference to `stepInfo` is being kept so `Name` property can point to `this._stepInfo.State.Name`